source/keymanweb.js

```diff
@@ -3542,16 +3572,102 @@
     {
       if(document.attachEvent)
         document.attachEvent('onfocusin', keymanweb._IEFocusIn);
-      else if(document.addEventListener)
-        document.addEventListener('DOMNodeInserted', keymanweb._DOMNodeInserted, true);
     }
    
     // Restore and reload the currently selected keyboard 
     keymanweb.restoreCurrentKeyboard(); 
 
+      /* Setup of handlers for dynamically-added and (eventually) dynamically-removed elements.
+       * Reference: https://developer.mozilla.org/en/docs/Web/API/MutationObserver
+       * 
+       * We place it here so that it loads after most of the other UI loads, reducing the MutationObserver's overhead.
+       */
+
+      keymanweb.observationTarget = document.querySelector('body');
+      keymanweb.mutationObserver = new MutationObserver(function(mutations) 
+        {
+          var dirtyFlag = false; // Notes if we need to recompute our .sortedInputs array.
+
+          for(i=0; i < mutations.length; i++)
+          {
+            mutation = mutations[i];
+            
+            for(j = 0; j < mutation.addedNodes.length; j++)
+            {
+              var addedNode = mutation.addedNodes[j];
+              var lcTagName = ""
+              
+              if(addedNode.tagName) {
+                lcTagName = addedNode.tagName.toLowerCase();
+              }
+              
+              var childAdditions = [];
+
+              // Will need to handle this in case of child elements in a newly-added element with child elements.
+              if(addedNode.getElementsByTagName) 
+              {
+                var arr = addedNode.getElementsByTagName('input');
+                for(k = 0; k < arr.length; k++)
```

Comment:
It almost works ... but we need to use `Array.from` to convert the `NodeList` that `getElementsByTagName` returns into an `Array`.  Except for, of course, thank you IE which means we can't use `Array.from`. Sorry I didn't check this properly before my earlier comment but the final is still cleaner. I think `arrayFromNodeList` belongs in `keymanweb.util`.

````
function arrayFromNodeList(nl) {
  var res=[];
  for(var i = 0; i < nl.length; i++) {
    res.push(nl[i]);
  }
  return res;
}

childAdditions = childAdditions.concat(
  arrayFromNodeList(addedNode.getElementsByTagName('input')),
  arrayFromNodeList(addedNode.getElementsByTagName('textarea')),
  arrayFromNodeList(addedNode.getElementsByTagName('iframe')),
);